anvill/src/TypeSpecification.cpp

```diff
@@ -723,7 +723,10 @@ TypeSpecification::ParseType(llvm::LLVMContext &llvm_context,
         return llvm::Type::getDoubleTy(llvm_context);
       case 'D':  // long double.
         i += 1;
-        return llvm::Type::getX86_FP80Ty(llvm_context);
+        // The type spec for floating point with size 10 and
+        // 12 bytes. Create a double type to avoid type mismatch
+        // for ST registers in remill
+        return llvm::Type::getDoubleTy(llvm_context);
```

Comment:
You need to check that this isn't embedded inside of a pointer or structure. If we have a pointer to a `long double`, or a structure containing a `long double` then I think we should leave it as is.

Reply:
Status on this @kumarak?

Reply:
It will check the spec index during parsing to verify if the spec is generated from the primitive types or is a part of the pointer/struct.